_docs/LitmusVagrantTesting.md

```diff
@@ -0,0 +1,55 @@
+---
+layout: page
+title: Testing Puppet Modules with Litmus and Vagrant
+---
+
+## Testing Puppet Modules with Litmus and Vagrant
+ 
+### The Setup 
+
+For this demo I'm gonna run through the entire process from start to finish - I hope that you find it useful.
+To start, the first thing that you need to do is clone down the module that you want to test; in this case we'll be using `puppetlabs-scheduled_task`. This can easily be accomplished with a simple command:
+
+```
+git clone https://github.com/user-name/puppetlabs-scheduled_task.git
+```
+
+Now that you have the module cloned down, simply run the below command to set up the environment:
+
+```
+bundle install --path .bundle/gems/
+```
+
+### Provisioning the Machine
+
+With the environment is setup it is time to provision the machine. Rather than spin up the vagrant machine directly, we can use Litmus as an intermediary to do it for us.
+
+![Litmus Vagrant Testing - provisioning the machine](Vagrant%20Provision.png)
+
+If you are testing against multiple different OSs then you can also utilise the `provision_list` command, which alows you to provision an entire suite of machines from pre-set lists found within the `provision.yml` file.
+
+With the machine provisioned, it's a simple matter of running the `install_agent` and `install_module` commands and you will be setup and ready to go. With `install_agent` you can run the command as it is to install a default version of the agent or you can specify puppet 5 or 6 specificilly.
+
+![Litmus Vagrant Testing - installing the agent and the module](Agent%20Module%20Install.png)
+
+### Running the Tests
+
+Now that the machine is fully setup it is time to run the tests.
+
+This can be done in several ways, either through Litmus or through rspec.
+
+We'll start with the command `bundle exec rake litmus:acceptance:parallel`.
+
+When this command is given, Litmus will retrieve each machine that has been setup during the provision step and run the test suite against them simultaneously. This is an excellent function for testing prior to a release or merge as it helps achieve full coverage in a fraction of the time that it would normally take.
+
+The next command that can be given is `bundle exec rake litmus:acceptance:127.0.0.1:55985`
+
+This command will run the test's only against the machine whose name was provided, allowing you to run your test suite against a specific OS. This can be useful when making a change specific to a certain OS or when re-running previously failed test's after applying a quick bugfix.
+
+The final command that we wll be covering is `TARGET_HOST=127.0.0.1:55985 bundle exec rspec spec/acceptance/should_create_task_spec.rb`.
+
+Rather than calling on Litmus this instead utilise the older, but still functional, rspec command. This can allow you to specify specific test files to be run rather than the entire suite as well as allowing more detailed debugs to be returned.
```

Comment:
```suggestion
Rather than calling on Litmus this instead utilises the older, but still functional, rspec command. This will allow you to specify specific test files to be run rather than the entire suite as well as allowing more detailed debugs to be returned.
```

Reply:
Changed to:
This allows